blog/_posts/2020-05-16-next-few-years.adoc

```diff
@@ -0,0 +1,193 @@
+
+I am very happy with test speed in rust-analyzer.
+One of my mistakes in IntelliJ is adding a lot of tests that use Rust standard library and are slow for that reason.
+In rust-analyzer, there's only three uber-integrated tests that need the real libstd, all others work from in-memory fixtures which contain only the relevant bits of std.
+
+But the build times leave a lot to be desired.
+And this is hugely important -- the faster you can build the code, the faster you can do everything else.
+Heck, even for improving build times you need fast build times!
+I was trying to do some compile-time optimizations in rust-analyzer recently, and measuring "is it faster to compile now?" takes a lot of time, so one has to try fewer different optimizations!
+
+The biggest problem here is that Rust, as a language, is hard to compile fast.
+One specific issue I hit constantly is that changing a deep dependency recompiles the world.
+This is in contrast to C/C++ where, if you don't touch a `.h` files, changing a dependency requires only re-linking.
+In theory, we can have something like this in Rust, by automatically deriving public headers from crates.
+Though I fear that without explicit, physical "this is ABI" boundary, it will be much less effective at keeping compile times under control.
+
+As an aside, if Rust stuck with `.crate` files, implementing IDE support would have been much easier :-)
+
+== Optimizing rustc Build
+
+Nonetheless, rust-analyzer is much easier to build than rustc.
+I believe there's a lot we can do for rustc build as well.
+
+I've written at length about this at length on https://internals.rust-lang.org/t/experience-report-contributing-to-rust-lang-rust/12012/17?u=matklad[irlo].
+The gist is that I think we can split rustc into front-end "text-processing" part, and backend "LLVM, linkers and real world" part.
+The front bit than could in theory be a bog standard Rust project, which doesn't depend on IO, external tools and C++ code at all.
+
+One wrinkle here is that rustc test suite at the moment consists predominantly of UI and run-pass tests integration, which work by building the whole compiler.
+Such test suite is ideal for testing conformance and catching regressions, but is not really well suited for rapid TDD.
+I think we should make an effort to build a unit test suite a-la rust-analyzer, such that it's easy, for example, to test name resolution without building the type checker, and which doesn't require stdlib.
+
+== Scaling Maintainance
+
+Finally, all changes here represent deep cuts into an existing body of software.
+Pushing such ambitious projects to completion require people, who can dedicate significant amounts of their time and energy.
+To put it bluntly, we need more dedicated folks working on the IDE tooling as a full time, payed job.
+I am grateful to my colleagues at https://ferrous-systems.com/[Ferrous Systems] who put a lot of energy into making this a reality.
```

Comment:
Add a usual "call to action" (consider ask your company to start supporting via opencollective)?
It's pity we don't have the "unsupportive support plan" on sale yet...